core/src/main/java/io/questdb/griffin/engine/ops/OperationFutureImpl.java

```diff
@@ -234,7 +234,11 @@ private int awaitWriterEvent(long timeout) throws SqlException {
                         default:
                             final int strLen = Unsafe.getUnsafe().getInt(event.getData() + Integer.BYTES);
                             final long strLo = event.getData() + 2L * Integer.BYTES;
-                            throw SqlException.$(tableNamePositionInSql, strLo, strLo + 2L * strLen);
+                            if (strLen == 0) {
+                                throw SqlException.$(tableNamePositionInSql, "writer command failed");
```

Comment:
this is a user facing message, imagine you execute "alter table add column" you get back "writer command failed". What is "writer", what is "command" ?

Reply:
Changed to `statement execution failed` in f2fdf7294dd3c08b887a904694d74e1a849d5c58 and added some error logging to report the error code.

Off-topic question: it seems that we have a lot of dead code in `TableSyncModel`. Is it a leftover?

Reply:
could be, but lets tidy this up on the WAL branch

Reply:
Sure thing.